examples/repo_example/basic_repo.py

```diff
@@ -89,8 +90,8 @@ def _in(days):
 
 # Define containers for role objects and cryptographic keys created below. This
 # allows us to sign and write metadata in a batch more easily.
-roles = {}
-keys = {}
+roles: Dict[str, Metadata] = {}
+keys: Dict[str, Dict[str, Any]] = {}
```

Comment:
Quick question, what is the criterion for a variable name to get type annotated?

Reply:
it's needed if mypy can't figure it out from the context (and mypy is not very advanced in this). So in this case `roles = { "root": root_md }` would have been fine but an empty container needs typing

Reply:
I think it does need annotation when you use containers and personally, I think it's better to define annotations when you use containers. It makes it so much easier to understand how the container should be used or is used later.